### ISSUE TITLE
[minor] controllers: clean up use of new.... s/xyz.NewXYZ()/xyz.New()

### ISSUE THREAD

**Issue by jayunit100** [kubernetes/kubernetes#15523]
**problem**

The controllers in each package under controller have redundant camel cased `New...` methods... like 
- `NewDaemonSetController` (could be just `daemon.New`).  
  -`NewJobController` (could be `job.New` )

for example

```
286         go daemon.NewDaemonSetsController(kubeClient, s.resyncPeriod).
```

and here again...

```
289         go job.NewJobController(kubeClient, s.resyncPeriod).
```

and so on...

```
290             Run(s.ConcurrentJobSyncs, util.NeverStop)
291
292         podautoscaler.NewHorizontalController(kubeClient, metrics.NewHeapsterMetricsClient(kubeClient)).
293             Run(s.HorizontalPodAutoscalerSyncPeriod)
294
```
#### I think the below seems the right way to do this..

Each controller **subpackage** has a `New(...)` function...  We know the controller by looking directly at the sub package.

```
295         deployment.New(kubeClient).
296             Run(s.DeploymentControllerSyncPeriod)
297     }
298
```

Shall we replace these all accordingly? It makes
- the use of controllers a lot easier to grep for 
- the code easier to reason about 
- slightly simplifies/reduces the amount of code (trivial)  
- It also reinforces 1-1 ratio  controller-to-package, which appears to be the convention we're following.

tl;dr?  If each controller is going to have its own package,  lets have the New() create methods be named exactly the same.

**Comment by smarterclayton** [comment]
Closing due to age, reopen if this is important.
